File: july15.cpp
//User function template for C++

class Solution
{
    public:
    //Function to delete middle element of a stack.
    void deleteMidValue(stack<int>&s, int sizeOfStack, int mid){
        if (sizeOfStack == mid){
            s.pop();
            return;
        }
        int temp = s.top();
        s.pop();
        deleteMidValue(s,s.size(),mid);
        s.push(temp);
    }
    void deleteMid(stack<int>&s, int sizeOfStack)
    {
        // code here.. 
        int mid = (int)(ceil((float)sizeOfStack/2));
        deleteMidValue(s,sizeOfStack,mid);
            
            
    }
};

File: july15.py
#User function Template for python3

class Solution:
    def deleteMidValue(self, s, sizeOfStack, mid):
        if sizeOfStack == mid:
            s.pop()
            return
        temp = s[-1]
        s.pop()
        self.deleteMidValue(s, len(s), mid)
        s.append(temp)

    def deleteMid(self, s, sizeOfStack):
        mid = (sizeOfStack + 1) // 2
        self.deleteMidValue(s, sizeOfStack, mid)

